src/vfs2.c

```diff
@@ -970,22 +975,42 @@ static struct wal_index_full_hdr initial_full_hdr(struct wal_hdr whdr)
 	return ihdr;
 }
 
-static void set_mx_frame(struct wal_index_full_hdr *ihdr,
+static void pgno_ht_insert(uint16_t *ht, size_t len, uint16_t fx, uint32_t pgno)
+{
+	uint32_t hash = pgno * 383;
```

Comment:
It can wrap but this is benign/intended---SQLite performs the multiplication in the same way: https://github.com/sqlite/sqlite/blob/f73912694bd1671a664e3455524fc2821681c920/src/wal.c#L1127